### PR TITLE
Add `SHOW FUNCTIONS` statement

### LINKED ISSUE #3386: Show all functions
**Summary**

Add a show command for all functions.

**Syntax**

```
SHOW FUNCTIONS  [LIKE 'pattern' | WHERE expr]
```

### PR DESCRIPTION
**Pull request by lianghanzhen**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add `SHOW FUNCTIONS` statement

## Changelog

- New Feature

## Related Issues

Fixes #3386

## Test Plan

Unit Tests

### CODE REVIEW

File: query/src/sql/sql_parser.rs [hunk 1]
```diff
@@ -229,6 +230,23 @@ impl<'a> DfParser<'a> {
                             Ok(DfStatement::ShowMetrics(DfShowMetrics))
                         } else if self.consume_token("USERS") {
                             Ok(DfStatement::ShowUsers(DfShowUsers))
+                        } else if self.consume_token("FUNCTIONS") {
```
Comment (BohuTANG) [inline query/src/sql/sql_parser.rs:233]: It would be nice move these code snippets to a funtion `parse_show_functions` like `parse_show_databases`

**Comment by BohuTANG** [comment]
/lgtm

Thank you @lianghanzhen .

Assume that `show functions` has 2 tasks:
1. `show functions` statement parser
2. `show functions` statement implemention

This PR is the task1 :D

**Comment by lianghanzhen** [comment]
@BohuTANG can you elaborate the 2nd task?

**Comment by lianghanzhen** [comment]
I think I need to add documentation for `SHOW FUNCTIONS`, I will send another PR.

**Pull request merged**